includes/Utilities/Tracker.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @package FacebookCommerce
+ */
+
+namespace SkyVerge\WooCommerce\Facebook\Utilities;
+
+defined( 'ABSPATH' ) or exit;
+
+/**
+ * Class for adding diagnostic info to WooCommerce Tracker snapshot.
+ *
+ * See https://woocommerce.com/usage-tracking/ for more information.
+ *
+ * @since %VERSION%
+ */
+class Tracker {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since %VERSION%
+	 */
+	public function __construct() {
+		add_filter(
+			'woocommerce_tracker_data',
+			array( $this, 'add_tracker_data' )
+		);
+	}
+
+	/**
+	 * Append our tracker properties.
+	 *
+	 * @param array $data The current tracker snapshot data.
+	 * @return array $data Snapshot updated with our data.
+	 * @since %VERSION%
+	 */
+	public function add_tracker_data( array $data = array() ) {
+		$connection_handler = facebook_for_woocommerce()->get_connection_handler();
+		if ( ! $connection_handler ) {
+			return $data;
+		}
+
+		if ( ! isset( $data['extensions'] ) ) {
+			$data['extensions'] = array();
+		}
+
+		$connection_is_happy = $connection_handler->is_connected() && ! get_transient( 'wc_facebook_connection_invalid' );
+		$data['extensions']['facebook-for-woocommerce']['is-connected'] = $connection_is_happy;
```

Comment:
I think there's a convention to use underscores in tracker keys, so I might change these to:

`$data['extensions']['facebook_for_woocommerce']['is_connected']`

Reply:
Or perhaps it's better to use something simpler like `$data['extensions']['facebook']['connected']`?

Reply:
Updated to the simpler `$data['extensions']['facebook']['connected']` for now. This means we're land-grabbing `[extensions][facebook]` but I think that's appropriate, since this is the primary Facebook extension for WooCommerce. 8f149a5dc88a73976b838f5433f0b5cdbdb0ac50

Reply:
(I had to rebase / recreate this PR and the aforementioned commit is no longer on this branch.)

Reply:
https://github.com/fbtransfer/facebook-for-woocommerce/pull/1817#discussion_r601667816